isis/src/qisis/objs/Footprint2DView/Footprint2DView.cpp

```diff
@@ -197,38 +208,48 @@ namespace Isis {
 
 
   /**
-   * Slot to connect to the itemAdded signal from the model. If the
-   * item is an image it adds it to the scene.
+   * Slot to connect to the itemAdded signal from the model. If the item is an image or shape it is 
+   * added to a list. When everything has been added, then the list is added to the scene through 
+   * signal/slot connection from ProjectItemProxyModel signal, itemsAdded which is connected to 
+   * this objects onItemsAdded slot. 
    *
    * @param[in] item (ProjectItem *) The item
    */
   void Footprint2DView::onItemAdded(ProjectItem *item) {
     if (!item || (!item->isImage() && !item->isShape())) {
       return;
     }
-    //TODO 2016-09-09 TLS  Handle Shapes-Create image from shape since qmos only handles images?
-    //                   Still don't know if shape should inherit from image or contain an image?
-    //
+
     Image *image;
-    ImageList images;
     if (item->isShape()) {
-      //TEMPORARY UNTIL SHAPE IS FIXED TO HOLD IMAGE, once Shape holds image go back to old code
-      // previous to 10-21-16
       image = new Image(item->shape()->cube());
     }
     else if (item->isImage()) {
       image = item->image();
     }
-    images.append(image);
-    m_sceneWidget->addImages(images);
-    m_fileListWidget->addImages(&images);
+
+    m_images.append(image);
```

Comment:
just out of curiosity, I guess we don't still need the m_sceneWidget and m_fileListWidget addImages calls?